syntax = "proto3";

package adenineio;

// The service definition.
service AdenineIo {
  rpc Sign (Request) returns (Response) {}
  rpc UploadAndSign (Request) returns (Response) {}
  rpc VerifyAndShow (Request) returns (Response) {}
}

// The request message
message Request {
  string api_key = 1;
  string input = 2;
}

// The response message
message Response {
  string output = 1;
  string status_message = 2;
  bool status = 3;
}